client-spark/spark2/src/main/java/org/apache/spark/shuffle/reader/RssShuffleReader.java

```diff
@@ -143,14 +142,14 @@ public RssShuffleReader(
       context.taskMetrics().incMemoryBytesSpilled(sorter.memoryBytesSpilled());
       context.taskMetrics().incDiskBytesSpilled(sorter.diskBytesSpilled());
       context.taskMetrics().incPeakExecutionMemory(sorter.peakMemoryUsedBytes());
+
       // Use completion callback to stop sorter if task was finished/cancelled.
-      Function1<TaskContext, Void> fn1 = new AbstractFunction1<TaskContext, Void>() {
```

Comment:
both 2.3 and 2.4 have these interfaces?

Reply:
Since 2.3 leaks the lambda style API, we use listener style API instead that both 2.3 and 2.4 support.